route/v1/types.go

```diff
@@ -369,14 +369,17 @@ type RouteIngressConditionType string
 const (
 	// RouteAdmitted means the route is able to service requests for the provided Host
 	RouteAdmitted RouteIngressConditionType = "Admitted"
-	// TODO: add other route condition types
+	// RouteDeprecated indicates that the route is using a deprecated configuration that may
+	// be incompatible with a future version of OpenShift. If this set to true on any route,
+	// the ingress operator may block upgrades.
+	RouteDeprecated RouteIngressConditionType = "Deprecated"
```

Comment:
The condition should be more assertive of the problem.  Deprecated is generally read in a "well someday I'll fix this" and we're actually highlight a security concern that we know will prevent this route from functioning in two versions.

`UnservableInFutureVersions` could carry a reason and message that we choose for each case where you need this case.  The reason and message should emphasize the insecure aspect of the route.

Reply:
Done.